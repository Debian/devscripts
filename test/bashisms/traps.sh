#!/bin/sh

trap foo ERR # BASHISM
trap foo RETURN # BASHISM
trap foo DEBUG # BASHISM

trap "echo BASHISM" ERR
trap "echo BASHISM" RETURN
trap "echo BASHISM" DEBUG

foo() {
    echo ": dummy function"
}

trap $(foo BASHISM) ERR
trap "$(foo BASHISM)" RETURN
trap "echo $foo BASHISM" DEBUG
