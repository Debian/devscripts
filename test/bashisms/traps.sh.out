possible bashism in bashisms/traps.sh line 3 (trap with ERR|DEBUG|RETURN):
trap foo ERR # BASHISM
possible bashism in bashisms/traps.sh line 4 (trap with ERR|DEBUG|RETURN):
trap foo RETURN # BASHISM
possible bashism in bashisms/traps.sh line 5 (trap with ERR|DEBUG|RETURN):
trap foo DEBUG # BASHISM
possible bashism in bashisms/traps.sh line 7 (trap with ERR|DEBUG|RETURN):
trap "echo BASHISM" ERR
possible bashism in bashisms/traps.sh line 8 (trap with ERR|DEBUG|RETURN):
trap "echo BASHISM" RETURN
possible bashism in bashisms/traps.sh line 9 (trap with ERR|DEBUG|RETURN):
trap "echo BASHISM" DEBUG
possible bashism in bashisms/traps.sh line 15 (trap with ERR|DEBUG|RETURN):
trap $(foo BASHISM) ERR
possible bashism in bashisms/traps.sh line 16 (trap with ERR|DEBUG|RETURN):
trap "$(foo BASHISM)" RETURN
possible bashism in bashisms/traps.sh line 17 (trap with ERR|DEBUG|RETURN):
trap "echo $foo BASHISM" DEBUG
